client/state/data-layer/wpcom/sites/comments-tree/index.js

```diff
@@ -25,24 +25,41 @@ export const fetchCommentsTreeForSite = ( { dispatch }, action ) => {
 			{
 				method: 'GET',
 				path: `/sites/${ siteId }/comments-tree`,
-				apiVersion: '1',
+				apiVersion: '1.1',
 				query: {
-					status: 'unapproved' === status ? 'pending' : status,
+					status,
 				},
 			},
 			action
 		)
 	);
 };
 
-const mapTree = ( tree, status, type ) =>
-	map( tree, ( [ commentId, postId, commentParentId ] ) => ( {
+const mapPosts = ( commentIds, postId ) => {
+	postId = parseInt( postId, 10 );
```

Comment:
I honestly thought we were using [ESLint's `no-param-reassign`](https://eslint.org/docs/rules/no-param-reassign), but apparently we aren't. 🤔 

It is my personal preference to conform to that rule, but eh, if it's not a constraint, then 🤷‍♂️

Reply:
yeah this is a very strong pattern we like to avoid in Calypso - or at least I know several of us do - never ever change the parameters.

personally I don't want to ever ever change anything, but some exceptions are fine.

I see the need here and I'd be almost fine with it other than I think that changing parameters does far more harm than changing a variable because we're just not used to parameters changing.

@kwight how would you feel about renaming the prop to something like `stringPostId` or `rawPostId` or `apiPostId` and then creating the new `const postId = parseInt( postId, 10 )`?

Reply:
I haven't actually changed any params; `mapTree` remains the same with `( tree, status, type )`. I just had to place `mapPosts` above (so that it's defined for `mapTree`), which makes the diff confusing 🙃

Reply:
actually we're changing the value of the `postId` param in line 39 above - I guess this was already the case in the file in which case okay fair, but might be a good small change while we're in here :)

Reply:
+1 for not overwriting the arg here. I'm actually curious why we need the `parseInt` here. If postId is `undefined` the new value may now be NaN, which is also falsey but may be unexpected.

Reply:
Oh! I understand now, yes, will update that.

We need the `parseInt` since the IDs are strings in the response – each post ID is now a key. But we'll want them as integers when adding to the store.